### PR TITLE
Update crossword-pdf-uploader to prepare for dual-running with new crosswords services

### PR DESCRIPTION
**Pull request by kenoir**
## What does this change?

Follows https://github.com/guardian/crossword-uploader/pull/34 to update and test the `crossword-pdf-uploader`. This change is in preparation for "dual-running" the old and new crossword services (that change will be in a future PR).

This change also moves some things around so the xml & pdf uploader have more similar structures and test coverage.

## How to test

- [x] Run the tests
- [x] Run this locally against the test environment with a PDF / upload with correct name
- [x] Deploy this to CODE and test it with a PDF / upload with correct name

## How can we measure success?

The `crossword-pdf-uploader` lambda is well tested and can be updated to "dual run" with the new service safely.

### CODE REVIEW

File: crossword-pdf-uploader/src/main/scala/com/gu/crossword/pdfuploader/CrosswordConfigRetriever.scala [hunk 1]
```diff
@@ -43,7 +43,15 @@ trait S3CrosswordConfigRetriever extends CrosswordConfigRetriever {
     val configFileKey = s"crossword-pdf-uploader/$stage/config.properties"
     val configInputStream = s3Client.getObject("crossword-uploader-config", configFileKey).getObjectContent
     val configFile: Properties = new Properties()
-    Try(configFile.load(configInputStream)) orElse sys.error("Could not load config file from s3. This lambda will not run.")
```
Comment (bryophyta) [inline crossword-pdf-uploader/src/main/scala/com/gu/crossword/pdfuploader/CrosswordConfigRetriever.scala]: out of interest: what's the reason for this change?

Comment (kenoir) [reply crossword-pdf-uploader/src/main/scala/com/gu/crossword/pdfuploader/CrosswordConfigRetriever.scala]: This follows a review from @andrew-nowak where he noted this was a bit non-idiomatic. 

If this fails, we just want to stop and not carry on and `orElse` returning something (but really exiting) is a bit surprising. `try {} catch {}` with the `sys.exit` is intended to be a bit clearer about what it's doing.

Comment (bryophyta) [reply crossword-pdf-uploader/src/main/scala/com/gu/crossword/pdfuploader/CrosswordConfigRetriever.scala]: got it, that makes perfect sense -- thanks!

**Review by bryophyta — APPROVED** [review]
Happy to approve, although with the caveat that I'm not confident I've understood all of the changes in depth because it's my first time looking at this codebase. If you'd like a more robust approval then I'm happy to chat about it/review together 👍
